File: MYRana/app/src/main/java/com/example/myrana/academy/AcademyQuestionBank.kt
package com.example.myrana.academy

/** بنك الأسئلة — مطابق لملف Python academy_game.py */
data class AcademyQuestion(
    val question: String,
    val options: List<String>,
    val answer: String,
)

enum class ChallengeType(val title: String) {
    MATH("➕ تحدي الرياضيات"),
    SCIENCE("🔬 تحدي العلوم"),
    LOGIC("🧩 تحدي الألغاز"),
}

object AcademyQuestionBank {

    val math = listOf(
        AcademyQuestion("كم ناتج 5 + 3؟", listOf("8", "6", "10"), "8"),
        AcademyQuestion("كم ناتج 9 - 4؟", listOf("5", "3", "7"), "5"),
        AcademyQuestion("كم ناتج 3 × 2؟", listOf("6", "5", "8"), "6"),
        AcademyQuestion("أي عدد أكبر؟", listOf("12", "9", "6"), "12"),
    )

    val science = listOf(
        AcademyQuestion("ما الكوكب الذي نعيش عليه؟", listOf("الأرض", "المريخ", "زحل"), "الأرض"),
        AcademyQuestion("ما العضو الذي يضخ الدم؟", listOf("القلب", "العين", "الأذن"), "القلب"),
        AcademyQuestion("ما رمز الماء؟", listOf("H2O", "O2", "CO2"), "H2O"),
        AcademyQuestion("النبات يحتاج إلى؟", listOf("ماء وضوء", "حديد فقط", "رمل فقط"), "ماء وضوء"),
    )

    val logic = listOf(
        AcademyQuestion("شيء نراه في الليل ولا نلمسه؟", listOf("القمر", "الكتاب", "القلم"), "القمر"),
        AcademyQuestion("له أسنان ولا يعض؟", listOf("المشط", "الكلب", "السمكة"), "المشط"),
        AcademyQuestion("كلما أخذت منه كبر؟", listOf("الحفرة", "الكأس", "الكتاب"), "الحفرة"),
        AcademyQuestion("ما الشيء الذي يمشي بلا رجلين؟", listOf("الوقت", "الكرسي", "الحجر"), "الوقت"),
    )

    fun forType(type: ChallengeType): List<AcademyQuestion> = when (type) {
        ChallengeType.MATH -> math
        ChallengeType.SCIENCE -> science
        ChallengeType.LOGIC -> logic
    }
}

data class CityBuilding(val emojiLabel: String, val name: String, val starCost: Int)

object AcademyCityCatalog {
    val buildings = listOf(
        CityBuilding("🏠 بيت المعرفة", "بيت المعرفة", 10),
        CityBuilding("🏫 مدرسة العلوم", "مدرسة العلوم", 20),
        CityBuilding("🌳 حديقة الذكاء", "حديقة الذكاء", 30),
        CityBuilding("🚀 مركز الفضاء", "مركز الفضاء", 40),
    )
}

File: myrana_flutter/lib/academy/academy_question_bank.dart
/// مطابق لـ AcademyQuestionBank.kt / academy_game.py
class AcademyQuestion {
  const AcademyQuestion({
    required this.question,
    required this.options,
    required this.answer,
  });

  final String question;
  final List<String> options;
  final String answer;
}

enum ChallengeType {
  math('➕ تحدي الرياضيات'),
  science('🔬 تحدي العلوم'),
  logic('🧩 تحدي الألغاز');

  const ChallengeType(this.title);
  final String title;
}

class AcademyQuestionBank {
  AcademyQuestionBank._();

  static const math = <AcademyQuestion>[
    AcademyQuestion(question: 'كم ناتج 5 + 3؟', options: ['8', '6', '10'], answer: '8'),
    AcademyQuestion(question: 'كم ناتج 9 - 4؟', options: ['5', '3', '7'], answer: '5'),
    AcademyQuestion(question: 'كم ناتج 3 × 2؟', options: ['6', '5', '8'], answer: '6'),
    AcademyQuestion(question: 'أي عدد أكبر؟', options: ['12', '9', '6'], answer: '12'),
  ];

  static const science = <AcademyQuestion>[
    AcademyQuestion(
      question: 'ما الكوكب الذي نعيش عليه؟',
      options: ['الأرض', 'المريخ', 'زحل'],
      answer: 'الأرض',
    ),
    AcademyQuestion(
      question: 'ما العضو الذي يضخ الدم؟',
      options: ['القلب', 'العين', 'الأذن'],
      answer: 'القلب',
    ),
    AcademyQuestion(
      question: 'ما رمز الماء؟',
      options: ['H2O', 'O2', 'CO2'],
      answer: 'H2O',
    ),
    AcademyQuestion(
      question: 'النبات يحتاج إلى؟',
      options: ['ماء وضوء', 'حديد فقط', 'رمل فقط'],
      answer: 'ماء وضوء',
    ),
  ];

  static const logic = <AcademyQuestion>[
    AcademyQuestion(
      question: 'شيء نراه في الليل ولا نلمسه؟',
      options: ['القمر', 'الكتاب', 'القلم'],
      answer: 'القمر',
    ),
    AcademyQuestion(
      question: 'له أسنان ولا يعض؟',
      options: ['المشط', 'الكلب', 'السمكة'],
      answer: 'المشط',
    ),
    AcademyQuestion(
      question: 'كلما أخذت منه كبر؟',
      options: ['الحفرة', 'الكأس', 'الكتاب'],
      answer: 'الحفرة',
    ),
    AcademyQuestion(
      question: 'ما الشيء الذي يمشي بلا رجلين؟',
      options: ['الوقت', 'الكرسي', 'الحجر'],
      answer: 'الوقت',
    ),
  ];

  static List<AcademyQuestion> forType(ChallengeType type) {
    switch (type) {
      case ChallengeType.math:
        return math;
      case ChallengeType.science:
        return science;
      case ChallengeType.logic:
        return logic;
    }
  }
}

class CityBuilding {
  const CityBuilding({
    required this.emojiLabel,
    required this.name,
    required this.starCost,
  });
  final String emojiLabel;
  final String name;
  final int starCost;
}

class AcademyCityCatalog {
  AcademyCityCatalog._();

  static const buildings = <CityBuilding>[
    CityBuilding(emojiLabel: '🏠 بيت المعرفة', name: 'بيت المعرفة', starCost: 10),
    CityBuilding(emojiLabel: '🏫 مدرسة العلوم', name: 'مدرسة العلوم', starCost: 20),
    CityBuilding(emojiLabel: '🌳 حديقة الذكاء', name: 'حديقة الذكاء', starCost: 30),
    CityBuilding(emojiLabel: '🚀 مركز الفضاء', name: 'مركز الفضاء', starCost: 40),
  ];
}
